src/scripts/party/evolutions/WeatherEvolution.ts

```diff
@@ -0,0 +1,41 @@
+class WeatherEvolution extends Evolution {
```

Comment:
I think we should keep the naming similar to other Evolution restrictions, and also include a level for this as it could just be set to 1 by default where it's not needed.
```suggestion
class WeatherRestrictedLevelEvolution extends Evolution {
```
Only other weather evolution I could find
![image](https://user-images.githubusercontent.com/7288322/99890796-e9f20b00-2cc7-11eb-8da6-0aa61dcc796c.png)